### PR TITLE
Implemented discovered host tests from UI with helper funtions and fixed existing tests

### PR DESCRIPTION
**Pull request by sghai**
Implemented following tests:
- test_positive_pxe_less_multi_nic_with_dhcp_unattended
- test_positive_pxe_multi_nic_unattended
- test_positive_refresh_facts_pxe_less
- test_positive_reboot
- test_positive_update_default_org

Fixed:
- test_positive_delete_multiple
- few other changes around `libvirt_discovery.py`

### CODE REVIEW

**Comment by sghai** [comment]
All tests are passed except one.. will look into failed one.
```console
robottelo]$ py.test tests/foreman/ui/test_discoveredhost.py::DiscoveryTestCase
============================================= test session starts ==============================================
platform linux2 -- Python 2.7.12, pytest-3.0.2, py-1.4.31, pluggy-0.3.1
rootdir: /home/sghai/QE/robottelo, inifile: 
collected 53 items 

tests/foreman/ui/test_discoveredhost.py .ssss.ss...ssssssssssssssssss.s.ssss.sss..s..ssFsssss


============================== 1 failed, 12 passed, 40 skipped in 2259.34 seconds ==============================
```

**Comment by sghai** [comment]
Fixed the failed test as well.
```console
robottelo]$ py.test -v tests/foreman/ui/test_discoveredhost.py::DiscoveryTestCase -k test_positive_update_default_org
============================================= test session starts ==============================================
platform linux2 -- Python 2.7.12, pytest-3.0.2, py-1.4.31, pluggy-0.3.1 -- /usr/bin/python
cachedir: .cache
rootdir: /home/sghai/QE/robottelo, inifile: 
collected 53 items 

tests/foreman/ui/test_discoveredhost.py::DiscoveryTestCase::test_positive_update_default_org <- robottelo/decorators/__init__.py PASSED
tests/foreman/ui/test_discoveredhost.py::DiscoveryTestCase::test_positive_update_default_org_from_settings <- robottelo/decorators/__init__.py SKIPPED

============================================= 51 tests deselected ==============================================
============================= 1 passed, 1 skipped, 51 deselected in 333.58 seconds =============================
```

**Comment by rochacbruno** [comment]
ACK

**Comment by sghai** [comment]
@oshtaier @abalakh: Hey, Please review the PR. Its pending since long.

**Comment by sghai** [comment]
Rebased

**Comment by rochacbruno** [comment]
@abalakh can you please review this PR? 
(or please if you can't can you you suggest someone else to review)

**Comment by renzon** [comment]
ACK

File: tests/foreman/ui/test_discoveredhost.py [hunk 1]
```diff
@@ -598,10 +673,28 @@ def test_positive_update_default_org(self):
...
+                    self._assertdiscoveredhost(host_2_name)
+                    hostnames = [host_1_name, host_2_name]
+                    for hostname in hostnames:
+                        host = self.discoveredhosts.search(hostname)
+                        if not host:
+                            raise UIError(
+                                'Could not find the selected discovered host '
+                                '"{0}"'.format(hostname)
+                            )
+                        self.discoveredhosts.navigate_to_entity()
```
Comment (abalakh) [inline tests/foreman/ui/test_discoveredhost.py]: Isn't this line redundant? `self.discoveredhosts.search()` will navigate to entity anyway

**Comment by abalakh** [comment]
Well, great. New github interface rules. While i was putting some comments PR got merged as new interface doesn't show the comments while review is finished :)

**Comment by abalakh** [comment]
And one more question - should this PR be merged to 6.2.z?

**Comment by omaciel** [comment]
@abalakh that is a great question for @sghai :)

**Comment by rochacbruno** [comment]
@abalakh good review, I think your comment should be addressed in a new PR :)

**Comment by sghai** [comment]
@abalakh: thank you for quick review. I'll surely address all your comments in next PR.  Also same changes are applicable for 6.2.z as well. Will raise a PR. thanks

**Pull request merged**